### ISSUE TITLE
Inherited Inner test classes causes fails when concrete class has no tests

### ISSUE THREAD

**Issue by akruijff** [bechte/junit-hierarchicalcontextrunner#27]
I wanted to inherit test and give each concrete class its own setup. This resulted in a exception that I included below. I've also included a code case. I am using Java 1.8. It seems HierarchicalContextRunner thinks that there is one more class than the really is.

```
@RunWith(HierarchicalContextRunner.class)
public class MyTest {
    public abstract class A {
        @Before
        public void setup() {}

        @Test
        public void runTestTwice() {}
    }

    public class B extends A {
        @Test
        public void runTestOnce() {}
    }

    public class C extends A {
    }
}
```

runTestTwice(org.mazarineblue.keyworddriven.MyTest$A)  Time elapsed: 0.003 sec  <<< ERROR!
java.lang.InstantiationException
    at sun.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:408)
    at de.bechte.junit.runners.util.ReflectionUtil.createDeepInstance(ReflectionUtil.java:85)
    at de.bechte.junit.runners.context.processing.MethodExecutor.run(MethodExecutor.java:76)
    at de.bechte.junit.runners.context.processing.MethodExecutor.run(MethodExecutor.java:57)
    at de.bechte.junit.runners.context.statements.RunChildren.evaluate(RunChildren.java:38)
    at de.bechte.junit.runners.context.statements.RunAll.evaluate(RunAll.java:27)
        ....

**Comment by bechte** [comment]
Unfortunately, there is no filter on `abstract inner classes`, see `ContextResolver.java` line 75:

```
protected boolean isAllowed(final Class<?> candidate) {
    return isNotStatic(candidate) && isPublic(candidate) && isNotIgnored(candidate);
}
```

I will add a filter for that case and add it into the next release. Thanks for reporting. :-)

**Comment by bechte** [comment]
fixed with commit f9ba943
